### ISSUE TITLE
Document decision tree about when to choose Az.Resources or Microsoft.Graph module for Azure AD user and group management

### ISSUE THREAD

**Issue by LeonarddeR** [Azure/azure-powershell#18248]
### Description

There are currently 4 PowerShell modules that can do user and group management in Azure AD:

* [MSOnline](https://docs.microsoft.com/en-us/powershell/module/msonline/?view=azureadps-1.0)
* [Azure AD PowerShell](https://docs.microsoft.com/en-us/powershell/azure/active-directory/overview?view=azureadps-2.0&preserve_view=true)
* [Microsoft Graph PowerShell](https://docs.microsoft.com/en-us/powershell/microsoftgraph/overview?view=graph-powershell-beta)
* AZ.Resources

I'm aware of the deprecation of the first two modules. Both point at Microsoft.Graph as a proper replacement, however I noticed that AZ.Resources also switched to Microsoft Graph lately. This all is pretty confusing. I think it should be properly documented when or not to choose for AZ.Resources instead of Microsoft.Graph, and when Microsoft.Graph should be preferred. If both modules serve a different audience, that should also be clear for people who need to make a decision for new projects and migration of old projects that use MSOnline or Azure AD PowerShell.

**Comment by dingmeng-xue** [comment]
Thanks for reporting. Due to Azure user needs to handle AzureAD user, group, application, and service principal with Azure services, Azure PowerShell provides their support in Az.Resources module. So, those four types are our focus.

MicrosoftGraph module serve all services behind MicrosoftGraph API. AzureAD is just a subset. So, if you want to manage objects in MicrosoftGraph API including AzureAD and others such as calendar, email, conversation and so on, MicrosoftGraph is a better choice.

If you only manage AzureAD object in your script, both can support your requirement.  

It's good suggestion that we should tell user in our conceptual doc.

**Comment by Keldandorin** [comment]
Yes please clear up the confusion and if possible keep to one model that rules it all. At the moment I can not find Invite guest user in either  AZ och MSGraph. Kinde hat using 2-3 modules and logon in them for every automation I do.

**Comment by MaddyMicrosoft** [comment]
Thanks for raising this. The same request, documenting when to use Az.Resources versus the Microsoft Graph PowerShell SDK for Azure AD / Entra object management, is now tracked in the more recent and more detailed #29005, so we are consolidating there and closing this as a duplicate.

On the specific point about inviting guest users, that is available today via New-MgInvitation in the Microsoft Graph PowerShell SDK (Microsoft.Graph.Identity.SignIns), which creates a B2B invitation. Az.Resources does not surface a guest-invite cmdlet, so Microsoft Graph is the module for that scenario.
